### ISSUE TITLE
java.lang.RuntimeException: Error initializing QuantumRenderer: no suitable pipeline found

### ISSUE THREAD

**Issue by ZhuLeYao** [nus-cs2103-AY2223S2/forum#220]
I am facing errors similar to #215 but this is already happening during the tutorial. 
However, the error only mentions  "Caused by: java.lang.RuntimeException: Error initializing QuantumRenderer: no suitable pipeline found"

I followed the tutorial for gradle and set up accordingly. I have also done the following:
1. changed my build.gradle application mainClassName to "duke.main.Launcher"
2. run > configurations > application changed build and run main class to duke.main.Launcher
3. removed main method from duke.java

StackOverflow mentioned either the modular path is pointed wrongly or I have the wrong javaFX downloaded e.g. 32bit or bit. However, I am using gradle so I did not download any javaFx and I believe my above is changing the modular path correctly. 

Thanks in advance.

**Comment by damithc** [comment]
@ZhuLeYao Some more info that can be useful:

* Which OS are you using?
* How are you running the application? Using Intellij or using the command `./gradelw run`?
* Give the code of your build.gradle and other relevant files (copy-paste code rather than screenshots, or push your code to a public repo and share the link here)

**Comment by ZhuLeYao** [comment]
@damithc 
- I am using macOS Ventura version 13.0.1
- Does this mean whether I am using Intellij or terminal commands? If so, I am running the application on Intellij, specifically the launcher file.

Here is my build.gradle:

```java
plugins {
    id 'java'
    id 'application'
    id 'com.github.johnrengelman.shadow' version '5.1.0'
}

repositories {
    mavenCentral()
}

dependencies {
    String javaFxVersion = '11'

    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'linux'

    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.5.0'
    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.5.0'
}

test {
    useJUnitPlatform()

    testLogging {
        events "passed", "skipped", "failed"

        showExceptions true
        exceptionFormat "full"
        showCauses true
        showStackTraces true
        showStandardStreams = false
    }
}

application {
    mainClassName = "duke.main.Launcher"
}

shadowJar {
    archiveBaseName = "duke"
    archiveClassifier = null
}

run{
    standardInput = System.in
}
```

Here is my Duke.java:
```java
package duke.main;

import java.io.*;

import javafx.application.Application;
import javafx.scene.Scene;
import javafx.scene.control.Label;
import javafx.stage.Stage;

/**
 * Runs the app.
 */
public class Duke extends Application {
    private Storage storage;
    private TaskList tasks;
    private Ui ui;
    private static final String filePath = "data/tasks.txt";

    /**
     * Initialises Storage, TaskList and Ui.
     * Previous tasks are loaded up.
     * Loading error will be shown if file cannot be found/ generated.
     *
     * @param filePath file path of text file containing tasks.
     */
    public Duke(String filePath) {
        ui = new Ui();
        storage = new Storage(filePath);
        try {
            tasks = new TaskList(storage.loadTxtFile());
        } catch (IOException e) {
            ui.showLoadingError(e.getMessage());
            tasks = new TaskList();
        }
    }

    //public static void main(String[] args) {
    //    new Duke("data/tasks.txt").run();
    //}

    /**
     * Runs the app.
     */
    public void run() {

        ui.printGreetingMessage();

        boolean saidBye = false;
        while (!saidBye) {
            String command = ui.getCommand();
            Parser parser = new Parser();
            parser.parse(command, ui, tasks, storage);
        }
    }

    @Override
    public void start(Stage stage) {
        Label helloWorld = new Label("Hello World!"); // Creating a new Label control
        Scene scene = new Scene(helloWorld); // Setting the scene to be our Label

        stage.setScene(scene); // Setting the stage to show our screen
        stage.show(); // Render the stage.
    }

}
```

Here is my Launcher.java:
```java
package duke.main;

import javafx.application.Application;
/**
 * A launcher class to workaround classpath issues.
 */
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
}
```

**Comment by damithc** [comment]
@ZhuLeYao not sure if related but do ensure you have followed the precise instructions for Mac users given in https://nus-cs2103-ay2223s2.github.io/website/admin/programmingLanguages.html

**Comment by ZhuLeYao** [comment]
The issue is settled but I would like to share the method.

Previously, I had brought up the issue in #1 of an error in my terminal instead of an unreadable GUI, which I thought was a different issue. The address book.jar works now according to the instructions. I have changed in intellij to zulu 11.0.18 for the below settings:
1. project structure
2. run>edit configurations
3. settings for maven runner
4. settings for gradle

However, now there is the error of "Caused by: java.lang.NoSuchMethodException: duke.main.Duke.<init>()"

<img width="1311" alt="Screenshot 2023-02-21 at 2 54 39 PM" src="https://user-images.githubusercontent.com/87946521/220270165-467a055c-2e06-4245-87ac-d581af562c74.png">

<img width="1318" alt="Screenshot 2023-02-21 at 2 54 30 PM" src="https://user-images.githubusercontent.com/87946521/220270200-25048fa0-7eb5-4ca7-9563-9acbfed4d83d.png">

<img width="1040" alt="Screenshot 2023-02-21 at 2 53 03 PM" src="https://user-images.githubusercontent.com/87946521/220270326-eeffe7eb-71ac-4beb-9123-4d305f802342.png">
<img width="939" alt="Screenshot 2023-02-21 at 2 53 40 PM" src="https://user-images.githubusercontent.com/87946521/220270332-33fb62e1-c5bb-40b4-b13c-268302c65b56.png">

From here, I saw past cs2103 forums that someone solved this by adding another duke constructor
public Duke() { }

This solved the issue finally even though I already had a Duke constructor.

Thank you @damithc for your help.
